docs/form-datasets.rst

```diff
@@ -138,29 +155,67 @@ The ``item`` blocks are analogous to rows in the CSV representation. Each ``item
+  :header: name, label, place, visits, last_visit, phone
 
-``instance("states")/root/item[name = ${my_state}]/population``
+  da0ee575-d…, Xue (Ifedore) - 2341745, c2139aae-5…, 1, 2024-5-1, 2341745
+  c51c32ac-1…, Dia (Ifedore) - 9868545, c2139aae-5…, 3, 2024-4-12, 9868545
 
-To get the total population across states with a population above a certain threshold:
+The components of lookup expressions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-``sum(instance("states")/root/item[population > ${pop_threshold}]/population)``
+.. image:: /img/form-datasets/lookup-expression.png
+
+Expressions to look up values in datasets always start with ``instance("<instance name>")`` to identify which dataset is being accessed. If you have a choice list named ``people`` or an attached CSV with filename ``people.csv``, your lookup expressions will start with ``instance("people")``.
+
+The next part of the expression is ``/root/item``. Each row in your choice list or attached file represents an **item**. ``/root/item`` selects all of those items.
 
 .. note::
 
-  You don't need to deeply understand the detail of these expressions to use them effectively. If you're interested in learning more, see the section on :ref:`XPath paths <xpath-paths>`. In particular, ``/root/item`` comes from the :ref:`XML structure used to represent datasets for selects <selects-from-xml>`. If you attach custom XML files to your form, you may have a different root node name.
+  You don't need to deeply understand the detail of these expressions to use them effectively. If you're interested in learning more, see the section on :ref:`XPath paths <xpath-paths>`. In particular, ``/root/item`` comes from the :ref:`XML structure used to represent datasets for selects <selects-from-xml>`. If you attach custom XML files to your form, this part of the expression may be different.
+
+You then generally need an expression in square brackets to filter down the list of items to only the ones you care about. For example, if you have a choice list or attached file named ``people`` and want to look up a value about a person that was selected in a form field called ``participant``, you would use ``name = ${participant}`` as your filter expression. ``name`` is the default property used to uniquely identify items in choices lists or Entity Lists. You can also filter the list by any other property. For example, see the image below for an example that filters a ``place`` list based on an ``enumerator`` property.
+
+.. note::
+
+  Most filter expressions you will write will use comparisons with ``=``. Once side of the ``=`` will be a property or column from your list, written without `${}`, and the other side will be a field from your form, written with `${}`. If you have more complex filtering needs, you can use any expression that evaluates to ``True`` or ``False``. These filter expressions are exactly the same kind of expression used to define a :ref:`choice filter <cascading-selects>`. 
```

Comment:
s/Once/One

Might be nice add "for example, `name=${participant}`" to make this easier to grok.